docs/usage/self-hosted-configuration.md

```diff
@@ -128,7 +128,16 @@ e.g.
 
 ## composerIgnorePlatformReqs
 
-Set to `false` to prevent usage of `--ignore-platform-reqs` in the Composer package manager.s
+By default, Renovate will run Composer with `--ignore-platform-reqs` as the PHP platform used by Renovate most probably won't match with the required PHP environment of your project as configured in your `composer.json` file.
+However, this also means that all platform constraints (including PHP version) will be ignored by default, which can result in updated dependencies that are not compatible with your platform.
+
+To solve this, you should configure explicit ignored platform requirements (for example `ext-zip`) by setting them separately in this array.
+Each item will be added to the Composer command with `--ignore-platform-req`, resulting in it being ignored during its invocation.
+
+If an empty array is configured, Renovate uses its default behaviour.
+
+Set to `null` to fully omit `--ignore-platform-reqs/--ignore-platform-req` during Composer invocation.
+This requires the Renovate image to be fully compatible with your Composer platform requirements in order for the Composer invocation to succeed.
```

Comment:
How can you see/know if your "Renovate image is fully compatible with the Composer platform requirements"?

Reply:
Trial and error I guess, you will see the problem soon enough as either the Composer invocation will fail completely, or you will see dependencies being updated that won't work in your other pipelines (one of the errors I currently see during my validation pipelines):

```
$ composer install --no-interaction --no-progress
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.
  Problem 1
    - psr/cache is locked to version 2.0.0 and an update of this package was not requested.
    - psr/cache 2.0.0 requires php >=8.0.0 -> your php version (7.3.27) does not satisfy that requirement.
  Problem 2
    - psr/cache 2.0.0 requires php >=8.0.0 -> your php version (7.3.27) does not satisfy that requirement.
    - doctrine/annotations 1.13.2 requires psr/cache ^1 || ^2 || ^3 -> satisfiable by psr/cache[2.0.0].
    - doctrine/annotations is locked to version 1.13.2 and an update of this package was not requested.
```

I would actually strongly recommend against using the `null` possibility for normal usage.

Reply:
@HonkingGoose Would like to see an adjustment here, or is it fine as it is now?

Reply:
If you recommend against using `null` in normal cases, we should definitely mention that.

Can you add a short explanation in the docs? I think what you had posted above can be edited to fit in?

Reply:
Okay, I did my best, I really believe this should be clear enough for any Composer user, otherwise I do not think they should bother using Renovate 😄